specification/redisenterprise/resource-manager/Microsoft.Cache/preview/2020-10-01-preview/redisenterprise.json

```diff
@@ -0,0 +1,1652 @@
+      "get": {
+        "tags": [
+          "PrivateLinkResources"
+        ],
+        "operationId": "PrivateLinkResources_ListByRedisEnterpriseCache",
+        "description": "Gets the private link resources that need to be created for a RedisEnterprise cluster.",
+        "x-ms-examples": {
+          "RedisEnterpriseListPrivateLinkResources": {
+            "$ref": "./examples/RedisEnterpriseListPrivateLinkResources.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved private link resources.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v1/privatelinks.json#/definitions/PrivateLinkResourceListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Sku": {
+      "description": "SKU parameters supplied to the create RedisEnterprise operation.",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The type of RedisEnterprise cluster to deploy. Possible values: (Enterprise_E10, EnterpriseFlash_F300 etc.)",
+          "enum": [
+            "Enterprise_E10",
+            "Enterprise_E20",
+            "Enterprise_E50",
+            "Enterprise_E100",
+            "EnterpriseFlash_F300",
+            "EnterpriseFlash_F700",
+            "EnterpriseFlash_F1500"
+          ],
+          "x-ms-enum": {
+            "name": "SkuName",
+            "modelAsString": true
+          }
+        },
+        "capacity": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The size of the RedisEnterprise cluster. Defaults to 2 or 3 depending on SKU. Valid values are (2, 4, 6, ...) for Enterprise SKUs and (3, 9, 15, ...) for Flash SKUs."
+        }
+      },
+      "required": [
+        "name"
+      ]
+    },
+    "ProvisioningState": {
+      "type": "string",
+      "readOnly": true,
+      "description": "Current status",
+      "enum": [
+        "Succeeded",
+        "Failed",
+        "Creating",
+        "Updating",
+        "Deleting",
+        "DeleteFailed",
+        "Disabled"
```

Comment:
The only allowed terminal states are Succeeded, Failed and Canceled. All other states will be treated as non-terminal by ARM. Also what is DeleteFailed supposed to represent?

Reply:
That is fine.  The Redis service has a bunch of other provisioning states as well besides Succeeded, Failed, and Cancelled, so I am confident Redis Enterprise can also have those other states.  DeleteFailed is supposed to represent a failed delete operation.

Reply:
No, it's not. This is how `provisioningState` must be modeled in ARM: https://github.com/Azure/azure-resource-manager-rpc/blob/423d048e58ce13cb8863fb2b2b22b4d6057c5318/v1.0/Addendum.md#provisioningstate-property. Terminal states that don't comply with that should be remapped to ones that do comply with the spec.

Reply:
@majastrz  So we should nix 'DeleteFailed' and use 'Failed' for it instead? Sounds good to me.

Reply:
Yes. You'd also need to do something with `Disabled`.

Reply:
If a user has an ARM resource which gets Disabled, what provisioningState is it supposed to be in?

Reply:
There is no corresponding value for that. This is typically modeled with a boolean `enabled` property or some sort of enum property that is separate from provisioning state.

Reply:
Yeah pretty much. `provisioningState` is used in a state machine inside ARM and it relies on the spec to determine if some operations finished (successfully or not). If ARM considers a particular state non-terminal, it may keep polling in certain cases or may even trigger failures in deployments or in other places.

I think your proposal makes sense. It keeps `provisioningState` as simple as possible and put all the detailed state info into a separate property.